src/tuneflow_py/models/song.py

```diff
@@ -83,6 +84,95 @@ def remove_track(self, track_id: str):
                 dep_track.remove_output()
         return track
 
+    def get_structures(self):
+        return self._proto.structures
+
+    def get_structure_at_index(self, index: int):
+        return self._proto.structures[index]
+
+    def get_structure_at_tick(self, tick: int):
+        def tickToStructureFn(tick):
+            return StructureMarker(song=self,tick=tick, type=StructureType.UNKNOWN)._proto
+
+        def structureToTickFn(structure):
+            return structure.tick
+
+        index = Song.get_structure_at_tick_impl(
+            tick, self._proto.structures, tickToStructureFn, structureToTickFn
+        )
+
+        if index < 0:
+            index = 0
+
+        if index >= len(self._proto.structures):
+            index = len(self._proto.structures) - 1
+
+        return self._proto.structures[index]
+
+    @staticmethod
+    def get_structure_at_tick_impl(
+            tick: int,
+            structures: List,
+            tickToStructureFn,
+            structureToTickFn,
+    ):
+        index = lower_equal(
+            structures,
+            tickToStructureFn(tick),
+            key=lambda s: structureToTickFn(s),
+        )
+
+        if index < 0:
+            index = 0
+
+        if index >= len(structures):
+            index = len(structures) - 1
+
+        return index
+
+    def create_structure(self, tick: int, type: StructureType):
+        structure = StructureMarker(song=self,tick=tick,type=type)
+        if len(self._proto.structures) == 1:
+            # If there is only 1 structure, move it to the start.
+            structure.set_tick(0)
+        self._proto.structures.append(structure._proto)
```

Comment:
Proto's append/insert/... methods creates a copy of your original proto instance, so you need to update your input wrapper's `_proto` to the newly inserted proto if you want to use your wrapper instance later.

```
self._proto.structures.append(structure._proto)
structure._proto = self._proto.structures[-1]
```

https://protobuf.dev/reference/python/python-generated/#repeated-message-fields